modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -731,12 +731,26 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
       assert(txRequest.transactions || txRequest.unsignedTxs, 'Unable to find transactions in txRequest');
       const unsignedTx =
         txRequest.apiVersion === 'full' ? txRequest.transactions![0].unsignedTx : txRequest.unsignedTxs[0];
-      await this.baseCoin.verifyTransaction({
-        txPrebuild: { txHex: unsignedTx.signableHex },
-        txParams: params.txParams || { recipients: [] },
-        wallet: this.wallet,
-        walletType: this.wallet.multisigType(),
-      });
+
+      // For ICP transactions, the HSM signs the serializedTxHex, while the user signs the signableHex separately.
+      // Verification cannot be performed directly on the signableHex alone. However, we can parse the serializedTxHex
+      // to regenerate the signableHex and compare it against the provided value for verification.
+      // In contrast, for other coin families, verification is typically done using just the signableHex.
+      if (this.baseCoin.getConfig().family === 'icp') {
```

Comment:
It's great if this works for both mainnet or testnet. But a better way to do this is something like -

```suggestion
import { CoinFamily } from '@bitgo-beta/statics';
      if (this.baseCoin.family === CoinFamily.ICP) {
  }
```